lesson24/src/js/main.js

```diff
@@ -73,4 +73,20 @@ submitButton.addEventListener("click", (e) => {
     window.location.href = "register-done.html";
 });
 
+const options = {
+    root: document.querySelector('[data-id="modal-inner"]'),
+};
+
+const removeDisabledForAgreeButton = (entries) => {
+    entries.forEach(entry => {
+        if(entry.isIntersecting){
+            const agreeButton  = document.querySelector('[data-id="agree_button"]');
+            agreeButton.removeAttribute('disabled');
+        }
+    });
+};
+
+const observer = new IntersectionObserver(removeDisabledForAgreeButton, options);
+observer.observe(document.querySelector('[data-id="last_text"]'));
+
```

Comment:
This is the part I added this PR.
After you have read through the modal's terms of use, the Disabled button in the modal will be removed from the Agree button so that you can click it. 
When you click the Agree button, the checkbox is automatically checked.